code/game/sound.dm

```diff
@@ -5,26 +5,27 @@
 
 	var/turf/turf_source = get_turf(source)
 
+	if(!turf_source)
+		return
+
 	//allocate a channel if necessary now so its the same for everyone
 	channel = channel || open_sound_channel()
 
  	// Looping through the player list has the added bonus of working for mobs inside containers
 	var/sound/S = sound(get_sfx(soundin))
 	var/maxdistance = (world.view + extrarange) * 3
-	var/list/listeners = GLOB.player_list
+	var/z = turf_source.z
+	var/list/listeners = SSmobs.clients_by_zlevel[z]
```

Comment:
I got a feeling this is gonna be the source of your runtime right here. You may want to debug print/debug out what the code THINKS is on that list or if it can event READ the list.